File: AOJ/0350/0350.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include<math.h>

int gcd(int x, int y){
  return y ? gcd(y, x%y) : x;
}

int print(int x, int y){
  int g = gcd(x, y);
  printf("%d/%d\n", x/g, y/g);
}

int main(){
  char in[10], all[10], sub[10];
  scanf("%s", in);
  int o, p, l, r, d, i, all_i = 0;
  p = -1;

  for ( i = 0; i < strlen(in); i++ ){
    if ( in[i] == '.' ){
      o = i;
    } else if ( in[i] == '(' ) {
      p = i;
      strncpy(sub, all, all_i);
      sub[all_i] = '\0';
    } else if ( in[i] == ')' ) {
    } else {
      all[all_i++] = in[i];
    }
  }
  all[all_i] = '\0';
  
  d = strlen(in) - o - 1;
  l = p - o - 1;
  if ( p == -1 ){
    print(atoi(all), pow(10, d));
  }else {
    d -= 2; // for ()
    print(atoi(all) - atoi(sub), pow(10, d) - pow(10, l) );
  }
  return 0;
}

File: AOJ/0350/0350.py
def gcd(x, y):
    return gcd(y, x % y) if y else x

def printV(x, y):
    g = gcd(x, y)
    print(str(x // g) + "/" +  str(y // g))

S = input()
o = S.find('.')
p = S.find('(')
d = len(S) - o - 1

if p == -1:
    all = S[0:o] + S[o + 1:len(S)]
    printV(int(all), 10**d)
else:
    sub = S[0:o] + S[o + 1:p]
    all = sub + S[p + 1:len(S) - 1]
    l = p - o - 1
    d -= 2 # for ()
    printV(int(all) - int(sub), 10**d - 10**l)
